File: LeetCode75/Binary_Tree_Level_Order_Traversal/Binary_Tree_Level_Order_Traversal.cc
// 102. Binary Tree Level Order Traversal C++
#include <vector>

// Definition for a binary tree node.
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
public:
    std::vector<std::vector<int>> levelOrder(TreeNode* root) {
        std::vector<std::vector<int>> v;
        if (!root) return v;

        std::vector<TreeNode*> level;
        level.emplace_back(root);

        std::vector<int> innerV;
        while (!level.empty()) {
            innerV.clear();
            std::vector<TreeNode*> tmp;
            for (auto x: level) {
                innerV.emplace_back(x->val);
                tmp.emplace_back(x->left);
                tmp.emplace_back(x->right);
            }
            v.emplace_back(innerV);
            level.clear();
            for (auto x:tmp) {
                if (x) {
                    level.emplace_back(x);
                }
            }
        }
        return v;
    }
};

File: LeetCode75/Binary_Tree_Level_Order_Traversal/Binary_Tree_Level_Order_Traversal.py
# 102. Binary Tree Level Order Traversal Python
# Definition for a binary tree node.
class TreeNode(object):
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution(object):
    def levelOrder(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
        if not root:
            return []
        
        output = []
        test = [root]
        while test:
            output.append([node.val for node in test])
            tmp = []
            for node in test:
                tmp.extend([node.left, node.right])
            test = [leaf for leaf in tmp if leaf]
        return output
        